### PR TITLE
[Doc] routine load support avro

### PR DESCRIPTION
**Pull request by hellolilyliuyi**
Fixes #issue

## What type of PR is this:
- [ ] BugFix
- [ ] Feature
- [ ] Enhancement
- [ ] Refactor
- [ ] UT
- [x] Doc
- [ ] Tool

## Checklist:
- [ ] I have added test cases for my bug fix or my new feature
- [ ] This pr will affect users' behaviors
- [ ] This pr needs user documentation (for new or modified features or behaviors)
  - [ ] I have added documentation for my new feature or new function

## Bugfix cherry-pick branch check:
- [x] I have checked the version labels which the pr will be auto-backported to the target branch
  - [x] 3.0
  - [ ] 2.5
  - [ ] 2.4
  - [ ] 2.3

### CODE REVIEW

File: docs/loading/RoutineLoad.md [hunk 1]
```diff
@@ -10,7 +10,7 @@ Routine Load supports data transformation at data loading and supports data chan
 
 ## Supported data formats
 
-Routine Load now supports consuming CSV and JSON format data from a Kafka cluster.
+Routine Load now supports consuming CSV, JSON and Avro (supported since v3.0.1) format data from a Kafka cluster.
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:13]: image
1) add a comma (,) preceding "and"?
2) change "format data" to "formatted data"?

File: docs/loading/RoutineLoad.md [hunk 2]
```diff
@@ -63,7 +63,7 @@ Routine Load now supports consuming CSV and JSON format data from a Kafka cluste
 
 ## Create a Routine Load job
 
-The following two examples describe how to consume CSV-format and JSON-format data in Kafka, and load the data into StarRocks by creating a Routine Load job. For detailed instruction and reference, see [CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE%20ROUTINE%20LOAD.md).
+The following three examples describe how to consume CSV-format, JSON-format and Avro-format data in Kafka, and load the data into StarRocks by creating a Routine Load job. For detailed instruction and reference, see [CREATE ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/CREATE%20ROUTINE%20LOAD.md).
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:66]: image
-> For detailed syntax and parameter descriptions?

File: docs/loading/RoutineLoad.md [hunk 3]
```diff
@@ -268,6 +268,128 @@ After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-re
...
+    "confluent.schema.registry.url"="http://172.xx.xxx.xxx:8081",  
+    "kafka_topic"="topic_0",  
+    "kafka_partitions"="0,1,2,3,4,5",  
+    "property.kafka_default_offsets"="OFFSET_BEGINNING"  
+);
+```
+
+- Data Format
+
+  You need to specify `"format"="avro"` in the clause `PROPERTIES` to define that the data format is Avro.
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:340]: image
If you add spaces preceding and following equation marks (=) in code snippets, do not forget to add spaces here.

File: docs/loading/RoutineLoad.md [hunk 4]
```diff
@@ -268,6 +268,128 @@ After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-re
...
+
+2. Register the Avro schema in the [Schema Registry](https://docs.confluent.io/platform/current/schema-registry/index.html).
+
+**Avro data**
+
+Prepare the Avro data and send it to the Kafka topic `topic_0`.
+
+#### Create a table
+
+According to the fields of Avro data, create a table `sensor_log` in the target database `sensor` in the StarRocks cluster. The column names of the table must match the field names in the Avro data. For the mapping of data types between the table columns and the Avro data fields, see [Data types mapping](#Data types mapping).
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:303]: image
suggest changing it to "the data type mapping"

File: docs/loading/RoutineLoad.md [hunk 5]
```diff
@@ -268,6 +268,128 @@ After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-re
...
+
+  > NOTE
+  >
+  > You do not need to specify the `COLUMNS` parameter if the names and number of the fields in the Avro record completely match those of columns in the StarRocks table.
+
+After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-reference/sql-statements/data-manipulation/SHOW%20ROUTINE%20LOAD.md) statement to check the status of the load job.
+
+#### Data types mapping
+
+StarRocks supports loading Avro data of all types. When Avro data is loaded into StarRocks, the data type mapping is as follows:
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:362]: image
I think this should mean -> The data type mapping between the Avro data fields you want to load and the StarRocks table columns is as follows:

File: docs/loading/RoutineLoad.md [hunk 6]
```diff
@@ -268,6 +268,128 @@ After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-re
...
+
+**Primitive type**
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:364]: Plural form is better

File: docs/loading/RoutineLoad.md [hunk 7]
```diff
@@ -268,6 +268,128 @@ After submitting the load job, you can execute the [SHOW ROUTINE LOAD](../sql-re
...
+| nul     | NULL      |
+| boolean | BOOLEAN   |
+| int     | INT       |
+| long    | BIGINT    |
+| float   | FLOAT     |
+| double  | DOUBLE    |
+| bytes   | STRING    |
+| string  | STRING    |
+
+**Complex type**
```
Comment (amber-create) [inline docs/loading/RoutineLoad.md:377]: Same as comment for Line 364. Plural form is better.

File: docs/sql-reference/sql-statements/data-manipulation/CREATE ROUTINE LOAD.md [hunk 8]
```diff
@@ -620,3 +625,235 @@ FROM KAFKA
...
+    `data_y` long NULL COMMENT "sensor data" 
+) 
+ENGINE=OLAP 
+PRIMARY KEY (id) 
+DISTRIBUTED BY HASH(`id`) BUCKETS 5; 
+```
+
+**Routine Load job**
+
+Submit the import job, use `jsonpaths` to specify the fields of the Avro data that need to be loaded. Note that for the subfield `data_y` in the nested Record, you need to specify its `jsonpath` as `"$.data.data_y"`.
```
Comment (evelynzhaojie) [inline docs/sql-reference/sql-statements/data-manipulation/CREATE ROUTINE LOAD.md:760]: import job---load job

**Review by amber-create — APPROVED** [review]

**Review by evelynzhaojie — APPROVED** [review]

**Comment by hellolilyliuyi** [comment]
@mergify backport branch-3.0

**Pull request merged**